### PR TITLE
[Feature] Add go-client compatible deserialization for activity tasks

### PR DESCRIPTION
**Pull request by 0x00101010**
## Intention
Add activity task deserialization logic that is compatible with official go implementation

### CODE REVIEW

File: lib/cadence/activity/task_processor.rb [hunk 1]
```diff
@@ -79,6 +79,24 @@ def respond_failed(reason, details)
...
+      def deserialize(input)
+        JSON.deserialize(task.input)
+      rescue Oj::ParseError
+        # cadence official go-client serializes / deserializes input in a different format than this ruby client
+        # adding additional deserialization logic here to help read input that is passed from go-client
+        #
+        # this ruby client serializes / deserializes everything as one big string like blow:
+        # [1012474654, "second input"]
+        #
+        # while go client serializes input as separate input followed by line break
```
Comment (haoxue1108) [inline lib/cadence/activity/task_processor.rb:92]: Can we link the go client serialization link here for reference?

Comment (0x00101010) [reply lib/cadence/activity/task_processor.rb:93]: done

**Review by gregwym — APPROVED** [review]

File: lib/cadence/activity/task_processor.rb [hunk 2]
```diff
@@ -79,6 +79,25 @@ def respond_failed(reason, details)
...
       end
+
+      def deserialize(input)
+        JSON.deserialize(task.input)
+      rescue Oj::ParseError
+        # cadence official go-client serializes / deserializes input in a different format than this ruby client
+        # adding additional deserialization logic here to help read input that is passed from go-client
+        # https://github.com/uber-go/cadence-client/blob/0.18.x/internal/encoding.go#L45-L58
+        #
+        # this ruby client serializes / deserializes everything as one big string like blow:
```
Comment (DeRauk) [inline lib/cadence/activity/task_processor.rb:90]: @0x00101010 nit: 'blow' -> 'below'

Comment (0x00101010) [reply lib/cadence/activity/task_processor.rb]: @DeRauk done

**Review by DeRauk — APPROVED** [review]

**Pull request merged**